### PR TITLE
feat(misc): owners claim with body, batch claim and export model

### PR DESCRIPTION
**Pull request by daddykotex**
Refs MM-6997, MM-7420, MM-8185

### CODE REVIEW

**Comment by daddykotex** [comment]
This would make the SDK on par with the python version 1.0.33

File: README.md [hunk 1]
```diff
@@ -317,17 +321,39 @@ OwnersSDK mnuboOwnersClient = mnuboClient.getOwnerClient();
 mnuboOwnersClient.claim( "john.smith@mycompany.com", "my_device_Id" );
 ```
 
+Batch claiming
+------------------
+This example describes how to link Objects to Owners:
+```
+OwnersSDK mnuboOwnersClient = mnuboClient.getOwnerClient();
+
+mnuboOwnersClient.batchClaim( Collections.singletonList(
```
Comment (mvonthron) [inline README.md]: `singletonList` with multiple elements?

File: README.md [hunk 2]
```diff
@@ -317,17 +321,39 @@ OwnersSDK mnuboOwnersClient = mnuboClient.getOwnerClient();
...
 mnuboOwnersClient.unclaim( "john.smith@mycompany.com", "my_device_Id" );
 ```
 
+Batch unclaiming
+------------------
+This example describes how to unlink Objects to Owners:
+```
+OwnersSDK mnuboOwnersClient = mnuboClient.getOwnerClient();
+
+mnuboOwnersClient.batchUnclaim( Collections.singletonList(
```
Comment (mvonthron) [inline README.md]: same thing

File: README.md [hunk 3]
```diff
@@ -317,17 +321,39 @@ OwnersSDK mnuboOwnersClient = mnuboClient.getOwnerClient();
...
+mnuboOwnersClient.batchClaim( Collections.singletonList(
+    new ClaimOrUnclaim("john.smith@mycompany.com", "my_device_Id"),
+    new ClaimOrUnclaim("freddy@mycompany.com", "my_device_Id2", Collections.singletonMap("x_timestamp", "2017-04-26T07:38:36+00:00")) 
+));
+```
+
 Unclaiming an object
 ------------------
-This example describes how to link an Object with an Owner:
+This example describes how to unlink an Object with an Owner:
```
Comment (mvonthron) [inline README.md]: "from" an owner maybe?

File: README.md [hunk 4]
```diff
@@ -800,6 +826,19 @@ boolean eventAExists = exists.get(UUID.fromString("22222222-2222-2222-2222-22222
 boolean eventBExists = exists.get(UUID.fromString("11111111-1111-1111-1111-111111111111"))
 ```
 
+Model
+---------
+
+The SDK allow you to retrieve the model in the current zone (production or sandbox)
```
Comment (mvonthron) [inline README.md]: allow*s* + full stop at the end.

File: README.md [hunk 5]
```diff
@@ -800,6 +826,19 @@ boolean eventAExists = exists.get(UUID.fromString("22222222-2222-2222-2222-22222
...
+
+This example describes how to send search query:
+```java
+MnuboSDKClient mnuboClient = MnuboSDKFactory.getClient( HOST , CONSUMER_KEY , CONSUMER_SECRET );
+Model model = mnuboClient.getModelClient().export();
+```
+
+See [mnubo documentation](https://smartobjects.mnubo.com/apps/doc/api_search.html?highlight=export#get-api-v3-model-export) for more information.
```
Comment (mvonthron) [inline README.md]: remove the "highlight" part from the URL

File: src/main/java/com/mnubo/java/sdk/client/models/ClaimOrUnclaim.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
+package com.mnubo.java.sdk.client.models;
+
+import com.mnubo.java.sdk.client.utils.ValidationUtils;
+
+import java.util.Collections;
+import java.util.Map;
+
+public class ClaimOrUnclaim {
```
Comment (mvonthron) [inline src/main/java/com/mnubo/java/sdk/client/models/ClaimOrUnclaim.java]: I'm not fan of having a data type with a name that says something and it's opposite at the same time. Can we find something neutral?

File: README.md (hunk 1 above)
Comment (daddykotex) [inline README.md]: that doc does not compile ;)

Comment (daddykotex) [inline README.md]: done

File: README.md (hunk 3 above)
Comment (daddykotex) [inline README.md]: done

File: src/main/java/com/mnubo/java/sdk/client/models/ClaimOrUnclaim.java (hunk 6 above)
Comment (daddykotex) [inline src/main/java/com/mnubo/java/sdk/client/models/ClaimOrUnclaim.java]: Help me out here ;)

File: README.md (hunk 4 above)
Comment (daddykotex) [inline README.md]: done

File: README.md (hunk 5 above)
Comment (daddykotex) [inline README.md]: done

File: README.md (hunk 2 above)
Comment (daddykotex) [inline README.md]: done

File: src/main/java/com/mnubo/java/sdk/client/models/ClaimOrUnclaim.java (hunk 6 above)
Comment (daddykotex) [inline src/main/java/com/mnubo/java/sdk/client/models/ClaimOrUnclaim.java]: Note that I used the same for .NET.

**Comment by daddykotex** [comment]
@dmasse76 if you have a minute

File: src/main/java/com/mnubo/java/sdk/client/models/ClaimOrUnclaim.java (hunk 6 above)
Comment (dmasse76) [inline src/main/java/com/mnubo/java/sdk/client/models/ClaimOrUnclaim.java]: Some ideas: `ObjectAndOwner`, `OwnedObject`.  Not sure they are better

File: src/main/java/com/mnubo/java/sdk/client/mapper/ModelDeserializer.java [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+
+    @Override
+    public Model deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException {
+        JsonNode root = jp.getCodec().readTree(jp);
+
+        Set<ObjectType> objectTypes = new HashSet<>();
+        Map<TypeKeyPair, ObjectAttribute> allObjectAttribute = new HashMap<>();
+
+        final JsonNode otNode = root.get("objectTypes");
+        if (otNode != null) {
```
Comment (dmasse76) [inline src/main/java/com/mnubo/java/sdk/client/mapper/ModelDeserializer.java]: Should we split into different private methods the extraction fo each entities.  That quite a long method.

**Comment by daddykotex** [comment]
@dmasse76 I'm confident it would not do the trick for model, because I mapped relations between object types and object attributes and event types and timeseries.

Maybe for `ClaimOrUnclaim` with `@JsonAnyGetter/@JsonAnySetter`, I'll have a look

**Comment by dmasse76** [comment]
@daddykotex If it's not a 1 to 1 mapping, then you need the custom serializer.

File: src/main/java/com/mnubo/java/sdk/client/mapper/ModelDeserializer.java (hunk 7 above)
Comment (daddykotex) [inline src/main/java/com/mnubo/java/sdk/client/mapper/ModelDeserializer.java]: done

**Comment by daddykotex** [comment]
@dmasse76 I could get away with it a `@JsonAnyGetter` :)

**Pull request merged**